src/index.js

```diff
@@ -0,0 +1,31 @@
+const _BedrockRealm = require('./bedrock/realm/Realm')
+const _BedrockAccount = require('./bedrock/account/Account')
+
+const _JavaRealm = require('./java/realm/Realm')
+const _JavaAccount = require('./java/account/Account')
+
+const { formatJavaAuth, formatBedrockAuth } = require('./util')
+
+const { BedrockRealmsRelyingParty } = require('./constants')
+
+const { Authflow } = require('prismarine-auth')
+
+class RealmAPI extends Authflow {
+  constructor (username, cache, options = {}, codeCallback) {
+    super(username, cache, { ...options, authTitle: false }, codeCallback)
+    this.auth = {
+      getJavaAuth: () => (options.test) ? (async () => ({}))() : this.getMinecraftJavaToken({ fetchProfile: true }).then(formatJavaAuth),
+      getBedrockAuth: () => (options.test) ? (async () => ({}))() : this.getXboxToken(BedrockRealmsRelyingParty).then(formatBedrockAuth)
+    }
+    this.Bedrock = {
```

Comment:
If the functional Realm API is the same (aside from endpoints/payload), why not expose one set of API? Why do we need to do the .Java or .Bedrock (and have an option in constructor to pick pc or bedrock type)?

Reply:
This was done with pauth in mind, exposing both APIs means switching between the two is slightly less inconvenient. Instead of creating 2 instances, we can have the one and cycle between which API is needed depending on what Realm the user is trying to connect to. Do you think it'd be better as a constructor option? I don't see much downside to leaving it as it Is.

Reply:
OK, so instead of extending prismarine-auth and exposing a duplicate API, I think the Realms API can take in a prismarine-auth Authflow instance based on the code I see here. This way both the libraries can be used together as necessary.

I'm still not sure why there needs to be two APIs for Java Edition/Bedrock Edition, if this is the case it seems like there isn't sufficient abstraction we're doing here. Ideally instead of just passing through whatever the server is returning, we should abstract the information away. This way if the backend API changes we do not need to propagate breaking API changes